components/policy/brave_browser_policy_provider_unittest.cc

```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/policy/brave_browser_policy_provider.h"
+
+#include <memory>
+
+#include "base/test/task_environment.h"
+#include "components/policy/core/common/policy_bundle.h"
+#include "components/policy/core/common/policy_namespace.h"
+#include "components/policy/core/common/policy_types.h"
+#include "components/policy/core/common/schema_registry.h"
+#include "components/prefs/testing_pref_service.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_policy {
+
+class BraveBrowserPolicyProviderTest : public ::testing::Test {
+ public:
+  BraveBrowserPolicyProviderTest() = default;
+  ~BraveBrowserPolicyProviderTest() override = default;
+
+  void SetUp() override {
+    pref_service_ = std::make_unique<TestingPrefServiceSimple>();
+    schema_registry_ = std::make_unique<policy::SchemaRegistry>();
+    provider_ = std::make_unique<BraveBrowserPolicyProvider>();
+  }
+
+ protected:
+  base::test::TaskEnvironment task_environment_;
+  std::unique_ptr<TestingPrefServiceSimple> pref_service_;
+  std::unique_ptr<policy::SchemaRegistry> schema_registry_;
+  std::unique_ptr<BraveBrowserPolicyProvider> provider_;
+};
+
+TEST_F(BraveBrowserPolicyProviderTest, Construction) {
+  // Provider should be constructed successfully
+  EXPECT_NE(nullptr, provider_);
```

Comment:
nit: won't construction always succeed?

Reply:
Dropped it completely in the latest commit and moved the second check into the test below it..